src/exchange_graph.cc

```diff
@@ -78,6 +78,7 @@ Arc::Arc(boost::shared_ptr<ExchangeNode> unode,
 Arc::Arc(const Arc& other)
     : unode_(other.unode()),
       vnode_(other.vnode()),
+      pref_(other.pref()),
```

Comment:
This might be interacting poorly with my recent determism-sorting changes. - possible causing segfaults like @mbmcgarry is seeing.  We should take a closer look at #1207 to make sure nothing unintended is now happening by adding this to Arc.